canister/src/api/send_transaction.rs

```diff
@@ -1,39 +1,45 @@
 use crate::{
-    charge_cycles, runtime, types::SendTransactionInternalRequest, verify_network, with_state,
-    with_state_mut,
+    charge_cycles, runtime,
+    types::{Flag, SendTransactionInternalRequest},
+    verify_network, with_state, with_state_mut,
 };
 use bitcoin::{consensus::Decodable, Transaction};
 use ic_btc_types::SendTransactionRequest;
 
 pub async fn send_transaction(request: SendTransactionRequest) {
-    verify_network(request.network.into());
-
-    charge_cycles(with_state(|s| {
-        s.fees.send_transaction_base
-            + s.fees.send_transaction_per_byte * request.transaction.len() as u128
-    }));
-
-    // Decode the transaction as a sanity check that it's valid.
-    let tx = Transaction::consensus_decode(request.transaction.as_slice())
-        .expect("Cannot decode transaction");
-
-    runtime::print(&format!("[send_transaction] Tx ID: {}", tx.txid()));
-
-    // Bump the counter for the number of (valid) requests received.
-    with_state_mut(|s| {
-        s.metrics.send_transaction_count += 1;
-    });
-
-    // Use the internal endpoint to send the transaction to the bitcoin network.
-    runtime::call_send_transaction_internal(
-        with_state(|s| s.blocks_source),
-        SendTransactionInternalRequest {
-            network: request.network.into(),
-            transaction: request.transaction,
-        },
-    )
-    .await
-    .expect("Sending transaction bitcoin network must succeed");
+    match with_state(|s| s.api_access) {
+        Flag::Disabled => panic!("`bitcoin_send_transaction` is disabled."),
+        Flag::Enabled => {
```

Comment:
nit: instead of nesting each method in a match, how about adding a `verify_api_access()` method? Then the code would look like this:

```
verify_api_access();
verify_network(request.network.into());
...
```